src/Symfony/Component/Cache/Adapter/TagInvalidationInterface.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Adapter;
+
+use Psr\Cache\CacheItemPoolInterface;
+use Psr\Cache\InvalidArgumentException;
+
+/**
+ * Interface for invalidating cached items using tag hierarchies.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+interface TagInvalidationInterface extends CacheItemPoolInterface
+{
+    /**
+     * Invalidates cached items using tag hierarchies.
+     *
+     * @param string $tag A tag hierarchy to invalidate.
+     *
+     * @return bool True on success.
+     *
+     * @throws InvalidArgumentException When $tag is not valid.
+     */
+    public function invalidate($tag);
```

Comment:
*Side: Also, might be a need to be able to expire by tags, as opposed to a removal liker here. Would be important if Cache component is to be used as backend for HTTPCache, with tagging ala FOSHttpCache.*